build/nodes-and-endpoints/join-mainnet.md

```diff
@@ -336,22 +336,61 @@ git checkout $BINARY_VERSION
 make build
 cp build/persistenceCore ~/.persistenceCore/cosmovisor/upgrades/$BINARY_VERSION/bin
 ```
-
 Now, at the upgrade height, **Cosmovisor** will upgrade swap the binaries.
 
-### Manual Option
+**2. Auto-download option**   
+Above manual option, still needs upgrade version binary to be manually added under cosmovisor/upgrades/upgrade-name/bin directory.
+For clarity, refer below folder layout for daemon home directory:-
+
+![alt text for screen readers](cosmo-visor-folder-layout.jpg)
+
+However, for people who don't need such control and want an automated setup to download binaries on all the validating nodes, there is an **auto-download** option.
+
+**NOTE**: We don't recommend using auto-download because it doesn't verify in advance if a binary is available. If there will be any issue with downloading a binary, the cosmovisor will stop and won't restart an App (which could lead to a chain halt). Therefore, it is every node operator's responsibility to analyze such risks and use cosmovisor as per requirement.
+
+#### Steps to enable the Auto-Download feature
+**Case A: When creating the governance proposal (skip if you're a node operator)**
```

Comment:
@ajeet97 to review this section please.

Reply:
The doc looks fine to me.

I have one question though,
Suppose we are running the chain with `cosmovisor` auto download feature on.
So what happens if we don't provide `--upgrade-info` in the proposal?

Does `cosmovisor` stops the chain when the proposal is passed?
Or does it behave the same way as if auto download was not turned on - meaning it will stop at upgrade height?

Reply:
Thanks @ajeet97 for raising this question. 
In this case, Cosmovisor will look for upgrade binary under daemon_home/cosmovisor/updrades dir first and if not found there it will try to download from upgrade-info JSON. If not specified there as well, then cosmovisor will not be able to restart with new version binary since its not available and yes chain will stop at upgrade height.  
I have tested possible scenarios including your testcase on local chain and below is the summary of observations:-

|     |     |     |     |     |
| --- | --- | --- | --- | --- |
| Test # | Is Auto-download feature flag enabled? (yes/no) | Is upgrade-info specified in proposal? (yes/no) | Is Upgrade version binary manually copied in the path cosmovisor/upgrades/..?  (yes/no) | chain upgrade status |
| 1   | YES | NO  | NO  | chain halts |
| 2   | YES | NO  | YES | chain upgraded from binary on disk |
| 3   | YES | YES | NO  | chain upgraded from binary on web |
| 4   | NO  | YES | NO  | chain halts since upgrades location not created under cosmovisor dir |